diff --git a/CMakeLists.txt b/CMakeLists.txt
index fad0497f..8e72cb34 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,9 +11,9 @@ project(liquid VERSION ${LIQUID_VERSION} LANGUAGES C CXX)
 
 # ---------------------------------------- options ----------------------------------------
 
-option(BUILD_EXAMPLES   "Set to ON to build examples"           ON)
-option(BUILD_AUTOTESTS  "Set to ON to build autotest program"   ON)
-option(BUILD_BENCHMARKS "Set to ON to build benchmark program"  ON)
+option(BUILD_EXAMPLES   "Set to ON to build examples"           OFF)
+option(BUILD_AUTOTESTS  "Set to ON to build autotest program"   OFF)
+option(BUILD_BENCHMARKS "Set to ON to build benchmark program"  OFF)
 option(ENABLE_SIMD      "Set to ON to build SIMD extensions"    ON)
 option(BUILD_SANDBOX    "Set to ON to build sandbox programs"   OFF)
 option(BUILD_DOC        "Set to ON to build documentation"      OFF)
diff --git a/include/liquid.h b/include/liquid.h
index 3db82f8b..f6066a26 100644
--- a/include/liquid.h
+++ b/include/liquid.h
@@ -9081,6 +9081,9 @@ int ofdmframesync_execute(ofdmframesync _q,
 // query methods
 float ofdmframesync_get_rssi(ofdmframesync _q); // received signal strength indication
 float ofdmframesync_get_cfo(ofdmframesync _q);  // carrier offset estimate
+void ofdmframesync_get_subcarrier_gain(ofdmframesync _q, 
+                            liquid_float_complex * _x,
+                            unsigned int _n); 
 
 // set methods
 int ofdmframesync_set_cfo(ofdmframesync _q, float _cfo);  // set carrier offset estimate
diff --git a/src/multichannel/src/ofdmframesync.c b/src/multichannel/src/ofdmframesync.c
index d15eeadc..08739391 100644
--- a/src/multichannel/src/ofdmframesync.c
+++ b/src/multichannel/src/ofdmframesync.c
@@ -448,6 +448,15 @@ float ofdmframesync_get_rssi(ofdmframesync _q)
     return -10.0f*log10(_q->g0);
 }
 
+// get receiver gain estimate
+void ofdmframesync_get_subcarrier_gain(ofdmframesync _q, 
+                            liquid_float_complex * _x,
+                            unsigned int _n)
+{
+ for (unsigned int i=0; i<_n; i++)
+        _x[i] = _q->G[i];
+};
+
 // get receiver carrier frequency offset estimate
 float ofdmframesync_get_cfo(ofdmframesync _q)
 {
